unstructured/ingest/connector/slack.py

```diff
@@ -0,0 +1,211 @@
+import json
+import os
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+
+from slack_sdk import WebClient
+from slack_sdk.errors import SlackApiError
+
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import (
+    requires_dependencies,
+    validate_date_args,
+)
+
+
+@dataclass
+class SimpleSlackConfig(BaseConnectorConfig):
+    """Connector config to process all messages by channel id."""
+
+    channel: str
+    token: str
+    oldest: str
+    latest: str
+
+    # Standard Connector options
+    download_dir: str
+    output_dir: str
+    re_download: bool = False
+    preserve_downloads: bool = False
+    verbose: bool = False
```

Comment:
there are more standard options needed here:
https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/connector/git.py#L30
...
(we'll fix this pattern soon)